sycl/doc/extensions/experimental/sycl_ext_oneapi_virtual_mem.asciidoc

```diff
@@ -0,0 +1,334 @@
+
+|`physical_mem(const device &syclDevice, const context &syclContext, size_t numBytes)` |
+Constructs a `physical_mem` instance using the `syclDevice` provided. This
+device must either be contained by syclContext or it must be a descendent device
+of some device that is contained by that context, otherwise this function throws
+a synchronous exception with the errc::invalid error code.
+
+This will allocate `numBytes` of physical memory on the device. `numBytes` must
+be a multiple of the minimum granularity, as returned by a call to
+`get_minimum_mem_granularity`
+
+|`physical_mem(const queue &syclQueue, size_t numBytes)` |
+Same as `physical_mem(syclQueue.get_device(), syclQueue.get_context, numBytes)`.
+
+|`void map(const void *ptr, size_t numBytes, size_t offset)` |
+Maps a virtual memory range, specified by `ptr` and `numBytes`, to the physical
+memory corresponding to the corresponding instance of `physical_mem`.
+
+The virtual memory range is inaccessible after this call and can be made
+accessible through a call to `set_access_mode`. Accessing an address in an
+inaccessible virtual memory range results in undefined behavior.
+
+|`void map(const void *ptr, size_t numBytes, size_t offset, access_mode mode)` |
+Maps a virtual memory range, specified by `ptr` and `numBytes`, to the physical
+memory corresponding to the corresponding instance of `physical_mem`.
+
+After this call the virtual memory range is accessible on the corresponding
+device in the access mode specified by `mode`. `mode` must either be
+`access_mode::read` or `access_mode::read_write`.
+
+Writing to any address in the virtual memory range with access mode set to
+`access_mode::read` results in undefined behavior.
+
+|`context get_context() const` |
+Returns the SYCL context associated with the instance of `physical_mem`.
+
+|`device get_device() const` |
+Returns the SYCL device associated with the instance of `physical_mem`.
+
+|`size_t size() const` |
+Returns the size of the corresponding physical memory in bytes.
+
+|============================
+
+Virtual memory address ranges are mapped to the a `physical_mem` through the
+`map` member functions. However, to unmap the virtual memory range the user
+only needs to know the context associated with the `physical_mem` the address
+range was mapped to. As such, the corresponding `unmap` is a free function
+defined as:
+
+```c++
+namespace sycl::ext::oneapi::experimental {
+
+void unmap(const void *ptr, size_t numBytes, const context &syclContext);
+
+} // namespace sycl::ext::oneapi::experimental
+```
+
+[frame="topbot",options="header,footer"]
+|=====================
+|Function |Description
+
+|`void unmap(const void *ptr, size_t numBytes, const device &syclDevice, const context &syclContext)` |
+Unmaps the range specified by `ptr` and `numBytes`. The range must have been
+mapped through a call to `physical_mem::map()` prior to calling this. The range
+must not be a proper sub-range of a previously mapped range.
```

Comment:
Description does't mention what the `syclContext` parameter does.

Reply:
I have added a note about the requirements for `syclContext`.